eo/src/eoRanking.h

```diff
@@ -40,73 +40,175 @@ template <class EOT>
+
+    // value() refers to the std::vector of worthes (we're in an eoParamvalue)
+    value().resize(pSize);
+
+    double beta = (2 - pressure) / pSize;
+    if (exponent == 1.0) // no need for exponetial then
+    {
+      double alpha = (2 * pressure - 2) / (pSize * pSizeMinusOne);
+      for (unsigned i = 0; i < pSize; i++)
+      {
+        int which = lookfor(rank[i], _pop);
+        value()[which] = alpha * (pSize - i) + beta; // worst -> 1/[P(P-1)/2]
+      }
+    }
+    else // exponent != 1
     {
-      std::vector<const EOT *> rank;
-      _pop.sort(rank);
-      unsigned pSize =_pop.size();
-      unsigned int pSizeMinusOne = pSize-1;
-
-      if (pSize <= 1)
-        throw eoPopSizeException(pSize,"cannot do ranking with population of size <= 1");
-
-      // value() refers to the std::vector of worthes (we're in an eoParamvalue)
-      value().resize(pSize);
-
-      double beta = (2-pressure)/pSize;
-      if (exponent == 1.0)         // no need for exponetial then
-        {
-          double alpha = (2*pressure-2)/(pSize*pSizeMinusOne);
-          for (unsigned i=0; i<pSize; i++)
-            {
-              int which = lookfor(rank[i], _pop);
-              value()[which] = alpha*(pSize-i)+beta; // worst -> 1/[P(P-1)/2]
-            }
-        }
-      else                                 // exponent != 1
-        {
-          double gamma = (2*pressure-2)/pSize;
-          for (unsigned i=0; i<pSize; i++)
-            {
-              int which = lookfor(rank[i], _pop);
-              // value in in [0,1]
-              double tmp = ((double)(pSize-i))/pSize;
-              // to the exponent, and back to [m,M]
-              value()[which] = gamma*pow(tmp, exponent)+beta;
-            }
-        }
+      double gamma = (2 * pressure - 2) / pSize;
+      for (unsigned i = 0; i < pSize; i++)
+      {
+        int which = lookfor(rank[i], _pop);
+        // value in in [0,1]
+        double tmp = ((double)(pSize - i)) / pSize;
+        // to the exponent, and back to [m,M]
+        value()[which] = gamma * pow(tmp, exponent) + beta;
+      }
     }
- private:
-  double pressure;	// selective pressure
+  }
+
+private:
+  double pressure; // selective pressure
   double exponent;
 };
 
+/**
+ * @class eoRankingCached
+ * @brief Cached version of eoRanking that stores precomputed values for better performance
+ *
+ * This class implements the same ranking algorithm as eoRanking but adds a caching layer
+ * that stores frequently used values when the population size remains constant between
+ * calls. This optimization is particularly useful in steady-state evolution where the
+ * population size typically doesn't change between selection operations.
+ *
+ * The caching mechanism stores:
+ * - Population size related values (pSize, pSizeMinusOne)
+ * - Precomputed coefficients (alpha, beta, gamma)
+ *
+ * Note: This optimization should only be used when the population size remains constant
```

Comment:
Doxygen allows making such a message more visible:
```suggestion
 * @warning This optimization should only be used when the population size remains constant
```